src/SmartVaults.ts

```diff
@@ -2056,48 +2057,58 @@ export class SmartVaults {
    *
    * @async
    * @param {string} policyId - The ID of the policy to which the label is to be associated.
-   * @param {SmartVaultsTypes.Label} label - The label object containing the data to be saved.
-   * @returns {Promise<SmartVaultsTypes.PublishedLabel>} - A promise that resolves to the published label.
+   * @param {SmartVaultsTypes.TransactionMetadata} label - The label object containing the data to be saved.
+   * @returns {Promise<SmartVaultsTypes.PublishedTransactionMetadata>} - A promise that resolves to the published label.
    * 
    * @throws {Error} - Throws an error if the policy event retrieval fails, or if shared keys are not found.
    * 
    * @example
    * const publishedLabel = await saveLabel('some-policy-id', { data: { 'Address': 'some-address' }, text: 'some-label-text' });
    */
-  async saveLabel(policyId: string, label: SmartVaultsTypes.Label): Promise<SmartVaultsTypes.PublishedLabel> {
+  saveLabel = async (policyId: string, label: SmartVaultsTypes.TransactionMetadata | Array<SmartVaultsTypes.TransactionMetadata>): Promise<Array<SmartVaultsTypes.PublishedTransactionMetadata>> => {
```

Comment:
I think we should update the name of this method to be saveTransactionMetadata, also code within the method still references label in various points that should be renamed, including the event kind